src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -38,6 +39,8 @@ using namespace epee;
 #include "crypto/hash.h"
 #include "ringct/rctSigs.h"
 
+#define PROBABILITY_MAKE_DUMMY_ADDITIONAL_TX_PUBKEY 0.2
```

Comment:
What do you think about this approach? I'm not entirely sure if the dummy additional tx pubkeys should be included always or at a certain probability (i.e. slightly more storage/processing cost vs potential statistical privacy leak). Maybe the added computational cost is negligible and we should favor privacy over everything else?